src/migrations/indexed-event-params.js

```diff
@@ -0,0 +1,141 @@
+const fs = require('fs-extra')
+const immutable = require('immutable')
+const path = require('path')
+const semver = require('semver')
+const toolbox = require('gluegun/toolbox')
+const yaml = require('yaml')
+
+const ABI = require('../abi')
+const Subgraph = require('../subgraph')
+
+const getGraphCliVersion = sourceDir => {
+  let pkgJsonFile = path.join(
+    sourceDir,
+    'node_modules',
+    '@graphprotocol',
+    'graph-cli',
+    'package.json',
+  )
+  let data = fs.readFileSync(pkgJsonFile)
+  let jsonData = JSON.parse(data)
+  return jsonData.version
+}
+
+const dataSourceHasEventHandlers = dataSource =>
+  dataSource &&
+  typeof dataSource === 'object' &&
+  ((dataSource.mapping &&
+    typeof dataSource.mapping === 'object' &&
+    dataSource.mapping.eventHandlers &&
+    Array.isArray(dataSource.mapping.eventHandlers) &&
+    dataSource.mapping.eventHandlers.length > 0) ||
+    (dataSource.templates &&
+      Array.isArray(dataSource.templates) &&
+      dataSource.templates.reduce(
+        (hasEventHandlers, template) =>
+          hasEventHandlers || dataSourceHasEventHandlers(template),
+        false,
+      )))
+
+const maybeUpdateEventSignature = (manifestFile, dataSource, event) => {
+  let abiName = dataSource.getIn(['source', 'abi'])
+  let abis = dataSource.getIn(['mapping', 'abis'], immutable.List())
+  let abi = abis.find(abi => abi.get('name') === abiName)
+  let name = abi ? abi.get('name') : undefined
+  let file = abi ? abi.get('file') : undefined
+
+  let oldEventSignatures
+  let newEventSignatures
+  if (file && name) {
+    try {
+      abi = ABI.load(name, path.relative(path.dirname(manifestFile), file))
+    } catch (e) {
+      // Failed to load the ABI, return the event unchanged
+      return event
+    }
+
+    oldEventSignatures = abi.oldEventSignatures()
+    newEventSignatures = abi.eventSignatures()
+  }
+
+  if (oldEventSignatures && newEventSignatures) {
+    for (let i = 0; i < oldEventSignatures.size; i++) {
+      // Only replace the event signature if the signature of the event has changed
+      // and the old signature is not present in the new signatures anymore.
```

Comment:
I don't get why this is necessary, why can't we just overwrite all signatures in the new form?

Reply:
It's not super obvious.

Let's say an event has more than one variant, e.g. one without indexed params and one with indexed params, e.g.
```yaml
SomeEvent(uint256)
SomeEvent(indexed uint256)
```

In the old manifest we'll find `SomeEvent(uint256)`. The question then is: do we translate this to `SomeEvent(indexed uint256)` or do we keep `SomeEvent(uint256)`. Both are present in the new set of signatures.

If `SomeEvent(uint256)` is no longer present, we can simply replace it. But if it's still around, this is a decision we can't really make on the users behalf because they could mean either of the variants. I feel in this case it's best to leave the signature untouched.

Reply:
I'll update the comment to include this explanation.

Reply:
Great explanation, thanks!